src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
```

Comment:
Ensure that the `MachineProducer` interface is parameterized properly to prevent creation of producers that do not use `Machine` or its subtype. As per the current implementation, it seems correct, but please double-check the `MachineProducer` interface definition to ensure it enforces the use of `Machine` or its subtypes.